### ISSUE TITLE
Save & Restore window locations

### ISSUE THREAD

**Issue by ScottBeeson** [digitalcreations/MaxTo#694]
**Is your feature request related to a problem? Please describe.**
Not that I know of

**Describe the solution you'd like**
I use a laptop.  90% of the time I am at my desk, with 2 additional monitors.  When I go to a meeting and return and plug everything back in, many of my windows have moved (not all).  

**Describe alternatives you've considered**
I've looked for other software solutions, but since I LOVE MaxTo and it performs a similar job, I thought it may be a good feature.

**Additional context**
I believe Windows is supposed to handle this but does not for everything.  I'm also thinking that a good feature for MaxTo would be to save the current regions AND windows as a template to restore or switch between.  Like I could have one profile for work and one for home that may have completely different regions and apps.

**Comment by vegardlarsen** [comment]
Thank you for your issue, @ScottBeeson. This comes up quite frequently, and its not something we will do. What you are experiencing is caused by a Windows video driver issue (as far as we can tell), and isn't really related to MaxTo (as in it happens on machines without MaxTo as well).

Remembering a window position isn't a hard problem, because each window has a window handle for as long as it is open. However, if you are talking about "I want Chrome to always go here when it starts", the problem is much more difficult, as there aren't any good identifiers to base it on:

- The **window handle** is different for each instance of a window. If you close the window, the window handle is invalid.
- The **window title** varies based on the content of the app. Chrome's window title will be something like "Save & Restore window locations - Issue #694 - digitalcreations/MaxTo - Google Chrome" if this tab is active, but if another tab is active it is completely different. 
- The **window class** is usually common for all windows in the same process (but it doesn't have to be, it varies from program to program).

That said, if what you are asking is something more like two recipe commands "Remember open window positions" and "Recall previous positions", this sounds easier and more reliable to implement. This would mean that it would only apply to any windows that were open at the time, and that are still open when recalling, would be affected. Think like the memory buttons on a calculator. Combined with triggers that would save the positions when Windows locks, and recalls them when you unlock the machine, could work (I am not 100% certain that it will, as we may miss the point where it is possible to get the positions of windows as Windows locks).

Is this something that would be interesting to you?

**Comment by ScottBeeson** [comment]
> What you are experiencing is caused by a Windows video driver issue (as far as we can tell), and isn't really related to MaxTo (as in it happens on machines without MaxTo as well).

Yeah, definitely not a MaxTo issue. I didn't mean to imply that if I did.

> Is this something that would be interesting to you?

Probably not. The way you are describing it it sounds like I would have to remember to click "remember open window positions" every time I need to unplug my laptop.

It's cool though. I'll try to see if there is some other utility that can do this.

**Comment by vegardlarsen** [comment]
> The way you are describing it it sounds like I would have to remember to click "remember open window positions" every time I need to unplug my laptop.

That's the part I am unsure about. We do have the ability to execute a recipe when Windows is locked (which I believe happens automatically before sleep), and I assume we can also set up some sort of trigger for sleep as well. The problem is that I am not sure if those points are already too late (e.g. the windows could have moved by then already).

It does warrant some further investigation, but I think it is a 50/50 shot if it can be done, and it would only work with the exact windows you have running when you "saved" the locations.

**Comment by sergonius** [comment]
I found MaxTo because FancyZones cannot maximize to the monitor, but they do save window positions in a way I'd expect. They only save the position if you set it to a zone, so to use your recipe example, I guess if you have the saving setting enabled, they call “remember” on setting the zone and “recall” on opening the window.

My case is just having different apps opening maximized on different monitors.

So in my case:

> Is this something that would be interesting to you?

Yes
